service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +39,266 @@ public void disbandTeam(JWTData userData, int teamId) {
+
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      GoalsRecord goal = db.newRecord(GOALS);
+      goal.setTeamId(teamId);
+      goal.setGoal(addGoalRequest.getGoal());
+      goal.setStartAt(addGoalRequest.getStart_at());
+      goal.setCompleteBy(addGoalRequest.getComplete_by());
+      goal.store();
+    } else {
+      throw new WrongTeamRoleException(teamId, usersTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void deleteGoal(JWTData userData, int teamId, int goalId) {
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      db.deleteFrom(GOALS).where(GOALS.ID.eq(goalId));
+    } else {
+      throw new WrongTeamRoleException(teamId, usersTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void inviteUser(JWTData userData, InviteUserRequest inviteUserRequest) {
+    UsersTeamsRecord inviterUserTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(inviteUserRequest.getTeamId()))
+            .fetchOne();
+
+    if (inviterUserTeamsRecord != null && inviterUserTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      inviteUserRequest
+          .getUsers()
+          .forEach(
+              (name, email) -> {
+                // Only invite a user if they are not a part of the team already
+                if (!db.fetchExists(
+                    USERS_TEAMS
+                        .USERS_TEAMS
+                        .join(USERS)
+                        .on(USERS.ID.eq(USERS_TEAMS.USER_ID))
+                        .where(
+                            USERS
+                                .EMAIL
+                                .eq(email)
+                                .and(TEAMS.ID.eq(inviteUserRequest.getTeamId()))))) {
+                  UsersRecord invitedUser =
+                      db.selectFrom(USERS).where(USERS.EMAIL.eq(email)).fetchOne();
+                  if (invitedUser != null) {
+                    UsersTeamsRecord usersTeams = db.newRecord(USERS_TEAMS);
+                    usersTeams.setTeamId(inviteUserRequest.getTeamId());
+                    usersTeams.setUserId(invitedUser.getId());
+                    usersTeams.setTeamRole(TeamRole.PENDING);
+                    usersTeams.store();
+                    // TODO SEND EMAIL
+                  }
+                }
+              });
+    } else {
+      throw new WrongTeamRoleException(
+          inviteUserRequest.getTeamId(), inviterUserTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public List<UsersTeamDataResponse> getApplicants(
+      JWTData userData, int teamId) {
+    List<UsersTeamsRecord> applicants =
```

Comment:
Check if team exists

Reply:
Can probably extract this out to a simple function that takes a team ID and run it at the start of a lot of these functions